src/libasr/ASR.asdl

```diff
@@ -173,10 +173,10 @@ stmt
         -- GoTo points to a GoToTarget with the corresponding target_id within
         -- the same procedure. We currently use `int` IDs to link GoTo with
         -- GoToTarget to avoid issues with serialization.
-    | GoTo(int target_id)
+    | GoTo(int target_id, identifier name)
```

Comment:
I think GoTo should just reference using an ID. The name should go into GoToTarget.
```suggestion
    | GoTo(int target_id)
```
Or am I missing something?

Reply:
Well in C backend we need the name to produce, `goto name_of_the_label`. So we need the name to be stored in both GoTo and GoToTarget.